std/functional.d

```diff
@@ -1068,6 +1068,146 @@ template memoize(alias fun, uint maxSize)
     }
 }
 
+/// ditto
+template memoize(alias fun, uint maxSize, bool lru = true)
+{
+    import std.traits : ReturnType;
+    import std.typecons : Tuple;
+
+    class Node
+    {
+    public:
+        Node next;
+        Node prev;
+        Tuple!(Parameters!fun) args;
+        ReturnType!fun res;
+
+        this(Tuple!(Parameters!fun) args = Tuple!(Parameters!fun).init,
+             ReturnType!fun res = ReturnType!fun.init)
+        {
```

Comment:
Hmmm, the defaults here are suprising - most functions will NOT return the default result for the default parameter values. Where are the defaults used?